extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Embeds an playbuzz item.
+ * The item-url attribute can be easily copied from a normal playbuzz URL.
+ * Example:
+ * <code>
+    <amp-playbuzz
+        item="http://www.playbuzz.com/perezhilton/poll-which-presidential-candidate-did-ken-bone-vote-for"
+        layout="responsive"
+        height="300"
+        width="300"
+        display-item-info="true"
+        display-share-buttons="true"
+        display-comments="true">
+    </amp-playbuzz>
+ * </code>
+ *
+ * For responsive embedding the width and height can be left unchanged from
+ * the example above and will produce the correct aspect ratio.
+ */
+
+import {CSS} from '../../../build/amp-playbuzz-0.1.css.js';
+import {logo} from './logo-image';
+import * as utils from './utils';
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {setStyles} from '../../../src/style';
+import {user} from '../../../src/log';
+import * as events from '../../../src/event-helper';
+import {whenDocumentComplete} from '../../../src/document-ready';
+import {postMessage} from '../../../src/iframe-helper';
+
+/** @const */
+const EXPERIMENT = 'amp-playbuzz';
+
+class AmpPlaybuzz extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+
+    /** @private {?string} */
+    this.item_ = '';
+
+     /** @private {?number} */
+    this.itemHeight_ = 300; //default
+
+     /** @private {?boolean} */
+    this.displayItemInfo_ = false;
+
+     /** @private {?boolean} */
+    this.displayShareBar_ = false;
+
+     /** @private {?boolean} */
+    this.displayComments_ = false;
+  }
+  /**
+   * @override
+   */
+  preconnectCallback() {
+    this.preconnect.preload(this.item_);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // EXPERIMENT
+    AMP.toggleExperiment(EXPERIMENT, true); //for dev
+    user().assert(isExperimentOn(this.win, EXPERIMENT),
+      `Enable ${EXPERIMENT} experiment`);
+
+    const e = this.element;
+    this.item_ = user().assert(
```

Comment:
no requirement for https, i just need it to be a valid playbuzz url.
a publisher can just copy his item url (http/https) and pase it into the amp-playbuzz src field and it should work.
that's the reason i'm dropping the scheme and use double slashes // for the beginning (to avoid cors issues)